includes/Modules/Analytics_4/Settings.php

```diff
@@ -37,6 +40,30 @@ public function register() {
 		$this->register_owned_keys();
 	}
 
+	/**
+	 * Gets Analytics 4 settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Analytics 4 settings, or default if not set.
+	 */
+	public function get() {
+		$value = parent::get();
+		$value = is_array( $value ) ? $value : $this->get_default();
```

Comment:
It's a minor consideration, but might using the [`option_`](https://developer.wordpress.org/reference/hooks/option_option/) filter be slightly better than overriding `get()`? This would ensure the value is applied when retrieving via other means and also would remove the need to manually retrieve the default if I've understood things correctly.

Reply:
Using a hook is the implicit approach to change something. I am not a fan on using hooks everywhere because it makes things much harder to track and understand. I like when we have things done explicitly in our API, whilst using hooks should be used as the last method when others don't work.

Reply:
Hmm - I do appreciate the point - and we are using `get()` as a means to override the settings value in some other cases, so it's in keeping with existing code.

It does of course mean that if we retrieve the value directly via `get_option()` this value won't be applied. Although we should avoid this in general usage, we do use `get_option()` in the unit tests. Just something to be aware of...